src/foam/util/SafetyUtil.java

```diff
@@ -59,6 +60,42 @@ public static int compare(Object o1, Object o2) {
     return ((Comparable) o1).compareTo(o2);
   }
 
+  public static int compare(FObject o1, FObject o2) {
+    if ( o1 == null && o2 == null ) return 0;
+    if ( o2 == null ) return  1;
+    if ( o1 == null ) return -1;
+    
+    return o1.compareTo(o2);
+  }
+
+  public static int compare(boolean o1, boolean o2) {
+    return o1 == o2 ? 0 : o1 ? 1 : 0;
+  }
+
+  public static int compare(String o1, String o2) {
+    return o1.equals(o2) ? 0 : o1 == null ? -1 : o2 == null ? 1 : o1.compareTo(o2);
```

Comment:
Check if o1 is null first. You check later, but you would have already have had a NPE by then.